modules/runner/src/integrationTest/java/org/apache/ignite/internal/ssl/ItSslTest.java

```diff
@@ -532,4 +512,65 @@ void jdbcCanConnectWithSslAndClientAuth() throws SQLException {
             }
         }
     }
+
+    @Test
+    @DisplayName("Cluster is not initialized when nodes are configured with incompatible ciphers")
+    void incompatibleCiphersNodes(@WorkDirectory Path workDir, TestInfo testInfo) {
+        Cluster cluster = new Cluster(testInfo, workDir);
+
+        String sslEnabledWithCipher1BoostrapConfig = createBoostrapConfig("TLS_AES_256_GCM_SHA384");
+        String sslEnabledWithCipher2BoostrapConfig = createBoostrapConfig("TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384");
+
+        CompletableFuture<IgniteImpl> node1 = cluster.startClusterNode(0, sslEnabledWithCipher1BoostrapConfig);
+        CompletableFuture<IgniteImpl> node2 = cluster.startClusterNode(1, sslEnabledWithCipher2BoostrapConfig);
+
+        String metaStorageAndCmgNodeName = testNodeName(testInfo, 0);
+
+        InitParameters initParameters = InitParameters.builder()
+                .destinationNodeName(metaStorageAndCmgNodeName)
+                .metaStorageNodeNames(List.of(metaStorageAndCmgNodeName))
+                .clusterName("cluster")
+                .build();
+
+        IgnitionManager.init(initParameters);
+
+        // First node will initialize the cluster with single node successfully since the second node can't connect to it.
+        assertThat(node1, willCompleteSuccessfully());
+        assertThat(node2, willTimeoutIn(1, TimeUnit.SECONDS));
```

Comment:
This will not work because the future completes only after the cluster is initialized. Well, maybe we could initialize the cluster consisting of one node and then add the second node.